src/lib/moduleMocker.js

```diff
@@ -47,6 +47,48 @@ function getType(ref) {
   return null;
 }
 
+function PlainConstructor() {}
+var plainObject = new PlainConstructor();
+/**
+ * methods on ES6 classes are not enumerable, so they can't
+ * be found with a simple `for (var slot in ...) {`
+ *
+ * scan will return an object with the same shape you would
+ * get with `for ... in` except the non-enumerable methods
+ * are included.
+ */
+ function scan(obj) {
```

Comment:
`getMethods` isn't really true. It also needs to return all enumerable props. [example](https://github.com/AsaAyers/jest/blob/es6-classes/src/lib/moduleMocker.js#L330)

It's kind of an inverted pollyfill. It allows a `for...in` loop over an ES6 class to work as if it were an ordinary object, or a CoffeeScript class.